src/topwidget.cpp

```diff
@@ -94,3 +98,40 @@ void TopWidget::mouseMoveEvent(QMouseEvent *event) {
     parentWidget()->move(delta);
     event->accept();
 }
+
+QToolButton* TopWidget::getBackButton() const
+{
+    auto app = KiwixApp::instance();
+    QAction *back = app->getAction(KiwixApp::HistoryBackAction);
+    return qobject_cast<QToolButton*>(widgetForAction(back));
+}
+
+QToolButton* TopWidget::getForwardButton() const
+{
+    auto app = KiwixApp::instance();
+    QAction *forward = app->getAction(KiwixApp::HistoryForwardAction);
+    return qobject_cast<QToolButton*>(widgetForAction(forward));
+}
+
+void TopWidget::updateBackForwardButtons()
+{
+    if (back_menu) {
+        back_menu->deleteLater();
+        back_menu = Q_NULLPTR;
+    }
```

Comment:
Do we really need to delete it ?
Who is the owner of the menu ? https://doc.qt.io/qt-5/qtoolbar.html#widgetForAction is not explicit about if the returned pointer is owned by the action or not.

Reply:
From doc on `void QToolButton::setMenu(QMenu *menu)`:
> Ownership of the menu is not transferred to the tool button.

QToolBar creates and owns QToolButton ("Back"), but this "Back" button doesn't own the history menu.